deployment/environment/devenv/don.go

```diff
@@ -377,6 +393,13 @@ func (n *Node) CreateJobDistributor(ctx context.Context, jd JobDistributor) (str
 		return "", err
 	}
 	// create the job distributor in the node with the csa key
+	resp, err := n.gqlClient.ListJobDistributors(ctx)
+	if err != nil {
+		return "", fmt.Errorf("Could not list job distrubutors: %w", err)
+	}
+	if len(resp.FeedsManagers.Results) > 0 {
+		return resp.FeedsManagers.Results[0].FeedsManagerParts.GetId(), nil
```

Comment:
I think this should be updated to check if the csakey matches with the JD, it should not just return the first one in the result